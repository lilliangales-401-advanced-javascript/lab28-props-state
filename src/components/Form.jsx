import React from "react";

export default class extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: this.props.name,
      count: this.props.count
    };
  }
  handleChange = event => {
    this.setState({ name: event.target.value }, () => {
      console.log();
    });
  };
  handleSubmit = event => {
    event.preventDefault();
    this.props.handleNameChange(this.state.name);
  };
  render() {
    return (
      <React.Fragment>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <button type="submit"> Change {this.props.name} name</button>

        </form>
      </React.Fragment>
    );
  }
}
