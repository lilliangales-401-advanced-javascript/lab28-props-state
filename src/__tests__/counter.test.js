import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../components/Form';
import Message from '../components/message';


describe("<Form />", () => {
  it('Form is rendered on the DOM', () => {
    let app = shallow(<Form />);
    expect(app.find('input').exists()).toBe(true);
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(<Form page="http://www.facebook.com">Facebook</Form>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Message />", () => {
  it('Form is rendered on the DOM', () => {
    let app = shallow(<Message />);
    expect(app.find('div').exists()).toBe(true);
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(<Form page="http://www.facebook.com">Facebook</Form>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});