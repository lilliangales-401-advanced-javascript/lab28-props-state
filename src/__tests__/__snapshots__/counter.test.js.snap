// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    type="submit"
  >
     Change 
     name
  </button>
</form>
`;

exports[`<Form /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    type="submit"
  >
     Change 
     name
  </button>
</form>
`;

exports[`<Message /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    type="submit"
  >
     Change 
     name
  </button>
</form>
`;
